crates/era-cheatcodes/tests/src/cheatcodes/ExpectRevert.t.sol

```diff
@@ -0,0 +1,195 @@
+        // we fail.
+        cheatcodes.expectRevert("revert");
+        reverter.doNotRevert();
+        reverter.revertWithMessage("revert");
+    }
+
+    function testExpectRevertConstructor() public {
+        cheatcodes.expectRevert("constructor revert");
+        new ConstructorReverter("constructor revert");
+    }
+
+    function testExpectRevertBuiltin() public {
+        Reverter reverter = new Reverter();
+        cheatcodes.expectRevert(abi.encodeWithSignature("Panic(uint256)", 0x11));
+        reverter.panic();
+    }
+
+    function testExpectRevertCustomError() public {
+        Reverter reverter = new Reverter();
+        cheatcodes.expectRevert(abi.encodePacked(Reverter.CustomError.selector));
+        reverter.revertWithCustomError();
+    }
+
+    function testExpectRevertNested() public {
+        Reverter reverter = new Reverter();
+        Reverter inner = new Reverter();
+        cheatcodes.expectRevert("nested revert");
+        reverter.nestedRevert(inner, "nested revert");
+    }
+
+    function testExpectRevertCallsThenReverts() public {
+        Reverter reverter = new Reverter();
+        Dummy dummy = new Dummy();
+        cheatcodes.expectRevert("called a function and then reverted");
+        reverter.callThenRevert(dummy, "called a function and then reverted");
+    }
+
+    function testDummyReturnDataForBigType() public {
+        Dummy dummy = new Dummy();
+        cheatcodes.expectRevert("reverted with large return type");
+        dummy.largeReturnType();
+    }
+
+    function testFailExpectRevertErrorDoesNotMatch() public {
+        Reverter reverter = new Reverter();
+        cheatcodes.expectRevert("should revert with this message");
+        reverter.revertWithMessage("but reverts with this message");
+    }
+
+    function testFailExpectRevertDidNotRevert() public {
+        Reverter reverter = new Reverter();
+        cheatcodes.expectRevert("does not revert, but we think it should");
+        reverter.doNotRevert();
+    }
+
+    function testExpectRevertNoReason() public {
+        Reverter reverter = new Reverter();
+        cheatcodes.expectRevert(bytes(""));
+        reverter.revertWithoutReason();
+    }
+
+    function testExpectRevertAnyRevert() public {
+        cheatcodes.expectRevert();
+        new ConstructorReverter("hello this is a revert message");
+
+        Reverter reverter = new Reverter();
+        cheatcodes.expectRevert();
+        reverter.revertWithMessage("this is also a revert message");
+
+        cheatcodes.expectRevert();
+        reverter.panic();
+
+        cheatcodes.expectRevert();
+        reverter.revertWithCustomError();
+
+        Reverter reverter2 = new Reverter();
+        cheatcodes.expectRevert();
+        reverter.nestedRevert(reverter2, "this too is a revert message");
+
+        Dummy dummy = new Dummy();
+        cheatcodes.expectRevert();
+        reverter.callThenRevert(dummy, "revert message 4 i ran out of synonims for also");
```

Comment:
```suggestion
        reverter.callThenRevert(dummy, "this as well is a revert message");
```

Reply:
Maybe we can add this upstream also :) https://github.com/matter-labs/foundry-zksync/blob/main/testdata/cheats/ExpectRevert.t.sol#L174